infra/modules/network/vpc-endpoints.tf

```diff
@@ -0,0 +1,101 @@
+locals {
+  # List of AWS services used by this VPC
+  # This list is used to create VPC endpoints so that the AWS services can
+  # be accessed without network traffic ever leaving the VPC's private network
+  # For a list of AWS services that integrate with AWS PrivateLink
+  # see https://docs.aws.amazon.com/vpc/latest/privatelink/aws-services-privatelink-support.html
+  #
+  # The database module requires VPC access from private networks to SSM, KMS, and RDS
+  aws_service_integrations = setunion(
+    # AWS services used by ECS Fargate: ECR to fetch images, S3 for image layers, and CloudWatch for logs
+    ["ecr.api", "ecr.dkr", "s3", "logs"],
+
+    # Feature flags with AWS Evidently
+    ["evidently", "evidently-dataplane"],
+
+    # AWS services used by the database's role manager
+    var.has_database ? ["ssm", "kms", "secretsmanager"] : [],
+  )
+
+  # S3 and DynamoDB use Gateway VPC endpoints. All other services use Interface VPC endpoints
+  interface_vpc_endpoints = toset([for aws_service in local.aws_service_integrations : aws_service if !contains(["s3", "dynamodb"], aws_service)])
+  gateway_vpc_endpoints   = toset([for aws_service in local.aws_service_integrations : aws_service if contains(["s3", "dynamodb"], aws_service)])
```

Comment:
Untested idea:

```suggestion
  # S3 and DynamoDB use Gateway VPC endpoints. All other services use Interface VPC endpoints
  gateway_services = ["s3", "dynamodb"]
  interface_vpc_endpoints = setsubtract(local.aws_service_integrations, local.gateway_services)
  gateway_vpc_endpoints   = setintersection(local.aws_service_integrations, local.gateway_services)
```

Reply:
I _**would**_ try that, but this file is a straight copy from the Nava template, and I kinda wanna keep it that way.

This is a good suggestion for the Nava template though! => https://github.com/navapbc/template-infra/blob/main/infra/modules/network/vpc-endpoints.tf